the-collab-lab/tcl-43-smart-shopping-list
# Rt ng sk refactor

## Description

In this PR we refactored our largest component `List` by creating a new `ListItem` component and extracting out repeated methods that were used by both of those components into the `utils.js` file.

## Related Issue

N/A

## Acceptance Criteria

N/A

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|    | :sparkles: New feature     |
|   ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

N/A - no UI changes

### After

N/A - no UI changes

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with `git fetch` followed by `git switch rt-ng-sk-refactor`
- Run `npm start` to launch the app
- Ensure previous functionality still working
